src/models/state/networking_state.rs

```diff
@@ -94,6 +94,17 @@ pub struct NetworkingState {
     /// record latest successful upgrade, merely latest attempt. This is to
     /// prevent excessive runs of the proof-upgrade functionality.
     pub last_tx_proof_upgrade_attempt: std::time::SystemTime,
+
+    /// Disconnection times of past peers. Can be used to determine if a connection
+    /// request should be accepted or rejected.
+    ///
+    /// Only records times of _graceful_ disconnections that were triggered by
+    /// _this_ node. That is, times of the following events are _not_ recorded:
+    /// - Graceful disconnect initiated by the peer.
+    /// - Abrupt disconnections, for example due to network failures.
+    ///
+    /// Only the peer tasks may update this map.
+    disconnection_times: HashMap<SocketAddr, SystemTime>,
```

Comment:
I think the key here should be `InstanceId`, not `SocketAddr`, since `SocketAddr` is unique for each new outgoing connection that a machine makes.